jenkins/attach_agent.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import ConfigParser
+import sys
+
+from jenkinsapi import jenkins  # sudo pip install jenkinsapi
+
+EXCLUSIVE = True
+SHARED = False
+
+INFO = {
+    'bespoke': ('bespoke', 1, EXCLUSIVE),
+    'heavy': ('build unittest', 2, EXCLUSIVE),
+    'light': ('node e2e', 20, EXCLUSIVE),
+    'pr': ('pull', 6, SHARED),
+}
+
+
+def info(host, kind):
+    labels, executors, exclusive = INFO[kind]
+    return {
+        'credential_description': 'Jenkins GCE ssh key',
+        'exclusive': exclusive,
+        'host': host,
+        'java_path': '',
+        'jvm_options': '',
+        'labels': labels,
+        'max_num_retries': 0,
+        'node_description': '',
+        'num_executors': executors,
+        'port': 22,
+        'prefix_start_slave_cmd': '',
+        'remote_fs': '/var/lib/jenkins',
+        'retry_wait_time': 0,
+        'suffix_start_slave_cmd': '',
+    }
```

Comment:
Unfortunately this recommendation isn't going to survive for the time being: it isn't exposed in the python api afaict